Python packages set: major updates

FRidh opened the pull request:
<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!
List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

###### Motivation for this change

It's been a long time since we did major updates in the whole set. Applications were not automatically updated, but likely need some fixing as well.

https://hydra.nixos.org/jobset/nixpkgs/python-unstable

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

FRidh committed:
python3Packages.mecab-python3: 1.0.2 -> 1.0.3

FRidh committed:
python3Packages.minio: 6.0.0 -> 6.0.2

FRidh committed:
python3Packages.nibabel: 3.2.0 -> 3.2.1

FRidh committed:
python3Packages.nbconvert: 5.6.1 -> 6.0.7

FRidh committed:
python3Packages.multidict: 4.7.6 -> 5.0.2

FRidh committed:
python3Packages.natsort: 7.0.1 -> 7.1.0

FRidh committed:
python3Packages.notify_py: 0.2.4 -> 0.3.1

FRidh committed:
python3Packages.nest_asyncio: 1.4.2 -> 1.4.3

FRidh committed:
python3Packages.notebook: 6.1.4 -> 6.1.5

FRidh committed:
python3Packages.more-itertools: 8.4.0 -> 8.6.0

FRidh committed:
python3Packages.networkx: 2.4 -> 2.5

FRidh committed:
python3Packages.nbdime: 2.0.0 -> 2.1.0

FRidh committed:
python3Packages.mlflow: 1.10.0 -> 1.12.1

FRidh committed:
python3Packages.neo: 0.8.0 -> 0.9.0

FRidh committed:
python3Packages.ntplib: 0.3.3 -> 0.3.4

FRidh committed:
python3Packages.numpy-stl: 2.12.0 -> 2.13.0

FRidh committed:
python3Packages.nunavut: 0.3.9 -> 0.6.2

FRidh committed:
python3Packages.packaging: 20.4 -> 20.7

FRidh committed:
python3Packages.panel: 0.9.7 -> 0.10.2

FRidh committed:
python3Packages.pathspec: 0.8.0 -> 0.8.1

FRidh committed:
python3Packages.pamqp: 2.3.0 -> 3.0.1

FRidh committed:
python3Packages.papermill: 2.1.3 -> 2.2.2

FRidh committed:
python3Packages.pdoc3: 0.9.1 -> 0.9.2

FRidh committed:
python3Packages.opentracing: 2.3.0 -> 2.4.0

FRidh committed:
python3Packages.parfive: 1.1.1 -> 1.2.0

FRidh committed:
python3Packages.pex: 2.1.20 -> 2.1.21

FRidh committed:
python3Packages.pbr: 5.4.5 -> 5.5.1

FRidh committed:
python3Packages.patator: 0.7 -> 0.9

FRidh committed:
python3Packages.numericalunits: 1.24 -> 1.25

FRidh committed:
python3Packages.outcome: 1.0.1 -> 1.1.0

FRidh committed:
python3Packages.portend: 2.6 -> 2.7.0

FRidh committed:
python3Packages.plyvel: 1.2.0 -> 1.3.0

FRidh committed:
python3Packages.pooch: 1.0.0 -> 1.3.0

FRidh committed:
python3Packages.preshed: 3.0.3 -> 3.0.4

FRidh committed:
python3Packages.plotly: 4.9.0 -> 4.13.0

FRidh committed:
python3Packages.pikepdf: 1.18.0 -> 2.2.0

FRidh committed:
python3Packages.prometheus_client: 0.8.0 -> 0.9.0

FRidh committed:
python3Packages.Pillow: 7.2.0 -> 8.0.1

FRidh committed:
python3Packages.phonenumbers: 8.12.12 -> 8.12.13

FRidh committed:
python3Packages.pglast: 1.12 -> 1.14

FRidh committed:
python3Packages.prettytable: 0.7.2 -> 2.0.0

FRidh committed:
python3Packages.pony: 0.7.13 -> 0.7.14

FRidh committed:
python3Packages.plaid-python: 7.1.0 -> 7.2.0

FRidh committed:
python3Packages.pip-tools: 5.3.1 -> 5.4.0

FRidh committed:
python3Packages.pkginfo: 1.5.0.1 -> 1.6.1

FRidh committed:
python3Packages.pybids: 0.12.3 -> 0.12.4

FRidh committed:
python3Packages.pykdtree: 1.3.1 -> 1.3.4

FRidh committed:
python3Packages.py3status: 3.28 -> 3.31

FRidh committed:
python3Packages.pyTelegramBotAPI: 3.7.3 -> 3.7.4

FRidh committed:
python3Packages.pyhaversion: 3.3.0 -> 3.4.2

FRidh committed:
python3Packages.pybtex: 0.22.2 -> 0.23.0

FRidh committed:
python3Packages.pygame: 1.9.6 -> 2.0.0

FRidh committed:
python3Packages.pygit2: 1.3.0 -> 1.4.0

FRidh committed:
python3Packages.pyfakefs: 4.1.0 -> 4.3.2

FRidh committed:
python3Packages.pygtrie: 2.3 -> 2.4.1

FRidh committed:
python3Packages.pycapnp: 0.6.4 -> 1.0.0

FRidh committed:
python3Packages.pyee: 7.0.4 -> 8.1.0

FRidh committed:
python3Packages.pydicom: 2.0.0 -> 2.1.1

FRidh committed:
python3Packages.prov: 1.5.3 -> 2.0.0

FRidh committed:
python3Packages.pyrsistent: 0.16.1 -> 0.17.3

FRidh committed:
python3Packages.pyserial-asyncio: 0.4 -> 0.5

FRidh committed:
python3Packages.pymongo: 3.11.0 -> 3.11.1

FRidh committed:
python3Packages.pymatgen: 2020.8.13 -> 2020.11.11

FRidh committed:
python3Packages.pyopencl: 2020.2.2 -> 2020.3.1

FRidh committed:
python3Packages.pymediainfo: 4.2.1 -> 5.0.3

FRidh committed:
python3Packages.pyro-api: 0.1.1 -> 0.1.2

FRidh committed:
python3Packages.pynisher: 0.5.0 -> 0.6.3

FRidh committed:
python3Packages.pyobjc: 6.2.2 -> 7.0

FRidh committed:
python3Packages.pylibacl: 0.5.4 -> 0.6.0

FRidh committed:
python3Packages.pyspread: 1.99.4 -> 1.99.5

FRidh committed:
python3Packages.pysonos: 0.0.35 -> 0.0.37

FRidh committed:
python3Packages.pyramid: 1.10.4 -> 1.10.5

FRidh committed:
python3Packages.pynput: 1.6.8 -> 1.7.1

FRidh committed:
python3Packages.pytest-flakes: 4.0.2 -> 4.0.3

FRidh committed:
python3Packages.pytest-mock: 3.2.0 -> 3.3.1

FRidh committed:
python3Packages.pytest-isort: 1.1.0 -> 1.2.0

FRidh committed:
python3Packages.pytest-flask: 1.0.0 -> 1.1.0

FRidh committed:
python3Packages.pytest-quickcheck: 0.8.5 -> 0.8.6

FRidh committed:
python3Packages.pytest-django: 3.9.0 -> 4.1.0

FRidh committed:
python3Packages.pytest-mpl: 0.11 -> 0.12

FRidh committed:
python3Packages.pysqueezebox: 0.4.0 -> 0.5.5

FRidh committed:
python3Packages.pytest-mypy: 0.6.2 -> 0.8.0

FRidh committed:
python3Packages.pytest-services: 2.1.0 -> 2.2.1

FRidh committed:
python3Packages.pytest-repeat: 0.8.0 -> 0.9.1

FRidh committed:
python3Packages.pytest-randomly: 3.4.1 -> 3.5.0

FRidh committed:
python3Packages.pytest-testmon: 1.0.2 -> 1.0.3

FRidh committed:
python3Packages.python-miio: 0.5.3 -> 0.5.4

FRidh committed:
python3Packages.python-rapidjson: 0.9.3 -> 0.9.4

FRidh committed:
python3Packages.python-fedora: 1.0.0 -> 1.1.1

FRidh committed:
python3Packages.pytest-xprocess: 0.14.0 -> 0.17.0

FRidh committed:
python3Packages.pytest-xdist: 1.34.0 -> 2.1.0

FRidh committed:
python3Packages.python-telegram-bot: 12.8 -> 13.0

FRidh committed:
python3Packages.python-redis-lock: 3.6.0 -> 3.7.0

FRidh committed:
python3Packages.python-markdown-math: 0.7 -> 0.8

FRidh committed:
python3Packages.python-periphery: 2.1.0 -> 2.1.1

FRidh committed:
python3Packages.python-socketio: 4.6.0 -> 4.6.1

FRidh committed:
python3Packages.pyxnat: 1.3 -> 1.4

FRidh committed:
python3Packages.pyusb: 1.0.2 -> 1.1.0

FRidh committed:
python3Packages.pytz: 2020.1 -> 2020.4

FRidh committed:
python3Packages.python-prctl: 1.7 -> 1.8.1

FRidh committed:
python3Packages.python-dotenv: 0.14.0 -> 0.15.0

FRidh committed:
python3Packages.s3fs: 0.4.2 -> 0.5.1

FRidh committed:
python3Packages.Routes: 2.4.1 -> 2.5.1

FRidh committed:
python3Packages.regex: 2020.7.14 -> 2020.11.13

FRidh committed:
python3Packages.robotframework-sshlibrary: 3.4.0 -> 3.5.1

FRidh committed:
python3Packages.r2pipe: 1.4.2 -> 1.5.3

FRidh committed:
python3Packages.restructuredtext_lint: 1.3.1 -> 1.3.2

FRidh committed:
python3Packages.qtconsole: 4.7.7 -> 5.0.1

FRidh committed:
python3Packages.rlp: 1.2.0 -> 2.0.1

FRidh committed:
python3Packages.readme_renderer: 26.0 -> 28.0

FRidh committed:
python3Packages.robotframework-requests: 0.7.1 -> 0.7.2

FRidh committed:
python3Packages.pyzmq: 19.0.2 -> 20.0.0

FRidh committed:
python3Packages.RestrictedPython: 5.0 -> 5.1

FRidh committed:
python3Packages.responses: 0.10.16 -> 0.12.1

FRidh committed:
python3Packages.ripser: 0.5.4 -> 0.6.0

FRidh committed:
python3Packages.skorch: 0.8.0 -> 0.9.0

FRidh committed:
python3Packages.sentry-sdk: 0.16.5 -> 0.19.4

FRidh committed:
python3Packages.sphinxcontrib-plantuml: 0.18.1 -> 0.19

FRidh committed:
python3Packages.sphinx_rtd_theme: 0.4.3 -> 0.5.0

FRidh committed:
python3Packages.sorl-thumbnail: 12.6.3 -> 12.7.0

FRidh committed:
python3Packages.scour: 0.38.1.post1 -> 0.38.2

FRidh committed:
python3Packages.sopel: 7.0.6 -> 7.0.7

FRidh committed:
python3Packages.sortedcontainers: 2.2.2 -> 2.3.0

FRidh committed:
python3Packages.setproctitle: 1.1.10 -> 1.2

FRidh committed:
python3Packages.slicer: 0.0.4 -> 0.0.5

FRidh committed:
python3Packages.spyder-kernels: 1.9.4 -> 1.10.0

FRidh committed:
python3Packages.sniffio: 1.1.0 -> 1.2.0

FRidh committed:
python3Packages.slither-analyzer: 0.6.13 -> 0.6.14

FRidh committed:
python3Packages.soco: 0.19 -> 0.20

FRidh committed:
python3Packages.testfixtures: 6.14.2 -> 6.15.0

FRidh committed:
python3Packages.terminado: 0.8.3 -> 0.9.1

FRidh committed:
python3Packages.stripe: 2.54.0 -> 2.55.0

FRidh committed:
python3Packages.spyder: 4.1.5 -> 4.2.0

FRidh committed:
python3Packages.staticjinja: 0.3.5 -> 0.4.0

FRidh committed:
python3Packages.sqlparse: 0.3.1 -> 0.4.1

FRidh committed:
python3Packages.sqlite-utils: 2.22 -> 3.0

FRidh committed:
python3Packages.strictyaml: 1.1.0 -> 1.1.1

FRidh committed:
python3Packages.sybil: 1.4.0 -> 2.0.0

FRidh committed:
python3Packages.typeguard: 2.9.1 -> 2.10.0

FRidh committed:
python3Packages.tornado: 6.0.4 -> 6.1

FRidh committed:
python3Packages.tqdm: 4.48.2 -> 4.54.0

FRidh committed:
python3Packages.tldextract: 2.2.3 -> 3.1.0

FRidh committed:
python3Packages.tifffile: 2020.8.25 -> 2020.11.26

FRidh committed:
python3Packages.traitsui: 7.0.1 -> 7.1.0

FRidh committed:
python3Packages.traitlets: 4.3.3 -> 5.0.5

FRidh committed:
python3Packages.tox: 3.19.0 -> 3.20.1

FRidh committed:
python3Packages.tvnamer: 3.0.1 -> 3.0.2

FRidh committed:
python3Packages.tflearn: 0.3.2 -> 0.5.0

FRidh committed:
python3Packages.tld: 0.12.2 -> 0.12.3

FRidh committed:
python3Packages.timezonefinder: 4.4.1 -> 4.5.0

FRidh committed:
python3Packages.transitions: 0.8.4 -> 0.8.5

FRidh committed:
python3Packages.west: 0.7.3 -> 0.8.0

FRidh committed:
python3Packages.uncertainties: 3.1.4 -> 3.1.5

FRidh committed:
python3Packages.virtualenv: 20.0.35 -> 20.2.1

FRidh committed:
python3Packages.uproot-methods: 0.8.0 -> 0.9.1

FRidh committed:
python3Packages.urllib3: 1.25.11 -> 1.26.2

FRidh committed:
python3Packages.uproot: 3.12.0 -> 3.13.1

FRidh committed:
python3Packages.vine: 1.3.0 -> 5.0.0

FRidh committed:
python3Packages.urwid_readline: 0.11 -> 0.12

FRidh committed:
python3Packages.vega_datasets: 0.8.0 -> 0.9.0

FRidh committed:
python3Packages.web.py: 0.61 -> 0.62

FRidh committed:
python3Packages.watchdog: 0.10.3 -> 0.10.4

FRidh committed:
python3Packages.ujson: 3.1.0 -> 4.0.1

FRidh committed:
python3Packages.validators: 0.17.1 -> 0.18.1

FRidh committed:
python3Packages.yt: 3.6.0 -> 3.6.1

FRidh committed:
python3Packages.XStatic-Pygments: 2.2.0.1 -> 2.7.2.1

FRidh committed:
python3Packages.wsproto: 0.15.0 -> 1.0.0

FRidh committed:
python3Packages.XStatic-jQuery-File-Upload: 9.23.0.1 -> 10.31.0.1

FRidh committed:
python3Packages.yarl: 1.5.1 -> 1.6.3

FRidh committed:
python3Packages.xcffib: 0.10.1 -> 0.11.0

FRidh committed:
python3Packages.xdot: 1.1 -> 1.2

FRidh committed:
python3Packages.zope.event: 4.4 -> 4.5.0

FRidh committed:
python3Packages.zm-py: 0.4.0 -> 0.5.2

FRidh committed:
python3Packages.zigpy-deconz: 0.9.2 -> 0.11.0

FRidh committed:
python3Packages.xml2rfc: 2.47.0 -> 3.5.0

FRidh committed:
python3Packages.zfec: 1.5.4 -> 1.5.5

FRidh committed:
python3Packages.zipp: 3.1.0 -> 3.4.0

FRidh committed:
python3Packages.XStatic-Bootstrap: 4.1.3.1 -> 4.5.3.1

FRidh committed:
python3Packages.zha-quirks: 0.0.46 -> 0.0.47

FRidh committed:
python3Packages.zope.interface: 5.1.2 -> 5.2.0

FRidh committed:
python3Packages.setuptools: 47.3.1 -> 50.3.1

FRidh committed:
python3Packages.wheel: 0.34.2 -> 0.35.1

FRidh committed:
python3Packages.pytest: 5 -> 6

FRidh committed:
fixup flit

FRidh committed:
python3Packages.pynacl: use latest hypothesis

jonringer committed:
python3Packages.email_validator: fix pname

jonringer committed:
python3Packages.codespell: fix tests

jonringer committed:
python3Packages.markdownsuperscript: mark broken

jonringer committed:
python3Packages.routes: fix tests

jonringer committed:
python3Packages.atlassian-python-api: fix tests

jonringer committed:
python3Packages.atpublic: fix tests

jonringer committed:
python3Packages.atsim_potentials: attempt fix, mark broken

jonringer committed:
python3Packages.atsim_potentials: fix homepage

jonringer committed:
python3Packages.ansi2html: fix build and tests

jonringer committed:
python37Packages.JPype1: fix build

jonringer committed:
python3Packages.block-io: fix build

jonringer committed:
python3Packages.colorlog: fix build

jonringer committed:
python3Packages.cx_Freeze: fix build

jonringer committed:
python3Packages.cx_Freeze: fix tests

FRidh commented:
numpy needs https://github.com/numpy/numpy/pull/17776

jonringer committed:
python37Packages.h2: fix tests

jonringer committed:
python37Packages.uvicorn: fix tests

jonringer commented:
the use of flit_core is making gimp fail because python2Packages.flit_core is disabled, which failing the eval as gimp is a darwin constituent https://gist.github.com/GrahamcOfBorg/f0aa782e418d7d426a7bf7a5a7372186

jonringer commented:
maybe we should revert the packaging bump until they figure out their existence.

jonringer committed:
python3Packages.isort: update homepage

jonringer committed:
python3Packages.isort: build from source, fix tests

jonringer committed:
python3Packages.pc-ble-driver-py: disable for python3.9

jonringer committed:
python3Packages.isort: refactor

jonringer committed:
python3Packages.databricks-connect: fix build

jonringer committed:
python3Packages.astroid: disable for python3.9

See https://github.com/PyCQA/astroid/issues/845

jonringer committed:
python3Packages.pyusb: fix build

jonringer committed:
python3Packages.pytest-services: fix build

jonringer committed:
python3Packages.uncompyle6: disable for python3.9

See: https://github.com/rocky/python-uncompyle6/issues/331

jonringer committed:
python3Packages.pooch: fix tests

jonringer committed:
python3Packages.keyrings-alt: fix build

jonringer committed:
python3Packages.inform: fix tests

jonringer committed:
python3Packages.numpy: use pytest_5

jonringer committed:
python3Packages.pynput: fix linux build

jonringer committed:
python37Packages.prettytable: fix build

jonringer committed:
python3Packages.jupyterlab-pygments: init at 0.1.2

jonringer committed:
python3Packages.nbclient: enable conditional checking

jonringer committed:
python3Packages.nbconver: fix tests

jonringer committed:
python3Packages.notebook: fix tests

jonringer committed:
python3Packages.pymatgen: fix deps

jonringer commented on pkgs/development/python-modules/nbconvert/default.nix:
```diff
@@ -29,21 +30,39 @@ buildPythonPackage rec {
+    "--ignore=nbconvert/tests/base.py"
+    "--ignore=nbconvert/tests/test_nbconvertapp.py"
+  ];
+
+
```
no

FRidh commented:
the use of flit_core is making gimp fail because python2Packages.flit_core is disabled, which failing the eval as gimp is a darwin constituent https://gist.github.com/GrahamcOfBorg/f0aa782e418d7d426a7bf7a5a7372186

In principle the Python 3 flit can make a wheel for Python 2 (they're universal wheels). I've been avoiding this but can use this fact.

FRidh committed:
python3Packages.re-assert: init at 1.1.0

FRidh committed:
python3Packages.aiohttp: fix build

FRidh committed:
python3Packages.numpy: fix compatibility with newer pytest

FRidh committed:
python3Packages.tqdm: fix build

FRidh committed:
Revert "python3Packages.panel: 0.9.7 -> 0.10.2"

Panel attempts to fetch js models from the web. Yikes!

This reverts commit 8d2cae0c2c4452aa0034e877540776b946a8374a.

FRidh committed:
python3Packages.poetry-core: isort is no test dep

FRidh committed:
pythonPackages.isort: fix build system, keep version for python 2

FRidh commented:
> In principle the Python 3 flit can make a wheel for Python 2 (they're universal wheels). I've been avoiding this but can use this fact.

I had a quick look at this approach. Fun stuff! So, in principle we could run the `flit` command, Python 3 version, to build the wheel. However, e.g. `packaging` uses PEP 517, so `pip` will attempt to use that method to find the backend. Then, it fails, because it cannot find the Python 3 version. Sigh.

Really the best approach here, if we were caring sufficiently for Python 2, would be to separate the wheel building part into a separate derivation. The installation function could then simply pick the universal wheel built with Python 3. Now I would still like to do that some day, and not for Python 2, but for other reasons, but not anytime soon.

So for now I suggest we keep the older packaging for Python 2 around.

FRidh committed:
python2Packages.packaging: keep 20.4 around

jonringer committed:
python3Packages.joblib: fix tests

jonringer committed:
python3Packages.jeepney: fix tests

jonringer committed:
python39Packages.boto: disable for python39

no longer compatible:
```
  File "/build/boto-2.49.0/tests/unit/utils/test_utils.py", line 88, in hmac_hashfunc
    return hmac.new(b'mysecretkey', msg)
  File "/nix/store/x3655l9x6b5khxzzl0s2zg1kn0qr3zfs-python3-3.9.0/lib/python3.9/hmac.py", line 170, in new
    return HMAC(key, msg, digestmod)
  File "/nix/store/x3655l9x6b5khxzzl0s2zg1kn0qr3zfs-python3-3.9.0/lib/python3.9/hmac.py", line 56, in __init__
    raise TypeError("Missing required parameter 'digestmod'.")
TypeError: Missing required parameter 'digestmod'.
```

jonringer committed:
python3Packages.h5py: fix build and disable tests

test suite now requires pytest-mpi which isn't available

jonringer committed:
python3Packages.pytest_xdist: use pytest_xdist_2 by default

meant to be used with pytest_6

jonringer committed:
libgit2: 1.0.0 -> 1.1.0

jonringer committed:
python3Packages.jsonrpc-websocket: fix tests

jonringer committed:
python3Packages.Pyro4: fix tests

jonringer committed:
python3Packages.portend: fix build

jonringer committed:
python3Packages.requests-unixsocket: fix tests

jonringer committed:
python3Packages.cheroot: fix tests by removing pytest-testmon

jonringer commented:
what do you want to do about:
> DEPRECATION: The -b/--build/--build-dir/--build-directory option is deprecated. pip 20.3 will remove support for this functionality. A possible replacement is use the TMPDIR/TEMP/TMP environment variable, possibly combined with --no-clean. You can find discussion regarding this at https://github.com/pypa/pip/issues/8333.

TMPDIR is already set during build, so we could just remove this option if I'm realing this correctly

jonringer committed:
python3Packages.secretstorage: 3.1.1 -> 3.3.0

jonringer committed:
python3Packages.pikepdf: fix tests

jonringer committed:
python3Packages.setproctitle: ignore impure tests

jonringer committed:
python3Packages.dask: fix tests

Tests cannot be ran concurrently

jonringer committed:
pgcli: allow for sqlparse 4.x

jonringer committed:
python3Packages.pyproj: fix tests

jonringer committed:
python3Packages.faker: 4.17.1 -> 4.18.0

jonringer committed:
python3Packages.faker: fix tests, remove random2

groodt commented:
> TMPDIR is already set during build, so we could just remove this option if I'm realing this correctly

That's the same interpretation I've taken here: https://github.com/NixOS/nixpkgs/pull/105540

FRidh committed:
pythonPackages.pip-install-hook: remove --build option

Pip had the option --build to build in a custom or temporary directory.
Nowadays, pip just listens to TMPDIR, which we already set.

This option was deprecated and is removed in pip 20.3.

FRidh committed:
python3Packages.pip: 20.2.4 -> 20.3

nixos-discourse commented:
This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:

https://discourse.nixos.org/t/upcoming-python-breakage/7554/19

FRidh commented:
Thank you @jonringer. There are about 900 failures left, which is about the same as it was before. I've updated to the latest pip.

TredwellGit commented:
After this was merged on staging:
```
$ nixos-rebuild build -I nixpkgs=.
building Nix...
building the system configuration...
error: pytest-xdist-2.1.0 not supported for interpreter python2.7
(use '--show-trace' to show detailed location information)
```

TredwellGit commented:
Seems to be caused by TeX Live depending on Python 2.

TredwellGit commented:
I have TeX Live working with Python 3; will request pull later.
